crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo

```diff
@@ -486,81 +491,1014 @@ mod YASPoolTests {
+                        maxLiquidityGross - 1000
+                    );
+            }
+
+
             #[test]
-            #[available_gas(2000000)]
-            fn test_fails_tick_lower_greater_than_tick_upper() {}
-        // TODO: 'fails if tickLower less than min tick'
-        // TODO: 'fails if tickUpper greater than max tick'
-        // TODO: 'fails if amount exceeds the max'
-        // TODO: 'fails if total amount at tick exceeds the max'
-        // TODO: 'fails if amount is 0'
+            #[available_gas(2000000000)]
+            #[should_panic(
+                expected: ('amount must be greater than 0', 'ENTRYPOINT_FAILED')
+            )] //set panic code
+            fn test_fails_amount_is_zero() {
+                let (yas_pool, token_0, token_1, mint_callback, min_tick, max_tick) = setup();
+                let (amount0, amount1): (u256, u256) = yas_pool
+                    .mint(
+                        recipient: yas_pool.contract_address,
+                        tick_lower: min_tick,
+                        tick_upper: max_tick,
+                        amount: 0,
+                        data: ArrayTrait::<felt252>::new()
+                    );
+            }
         }
 
+
         mod SuccessCases {
-            // TODO: 'initial balances'
-            // TODO: 'initial tick'
+            use super::{
+                setup, MIN_TICK, MAX_TICK, tick_spacing, FeeAmount, fee_amount,
+                IERC20DispatcherTrait, FACTORY_ADDRESS, TOKEN_A, TOKEN_B, WALLET, STATE,
+                encode_price_sqrt_1_1
+            };
+            use super::super::{get_min_tick_and_max_tick};
+            use yas_core::numbers::signed_integer::{
+                i32::i32, i32::i32_div_no_round, integer_trait::IntegerTrait
+            };
+            use yas_core::contracts::yas_pool::{
+                YASPool, YASPool::ContractState, YASPool::InternalImpl, IYASPool,
+                IYASPoolDispatcher, IYASPoolDispatcherTrait
+            };
+            use yas_core::contracts::yas_router::{
+                YASRouter, IYASRouterDispatcher, IYASRouterDispatcherTrait
+            };
+
+            #[test]
+            #[available_gas(200000000)]
+            fn test_initial_balances() {
+                let (yas_pool, token_0, token_1, mint_callback, min_tick, max_tick) = setup();
+
+                let balance_token_0 = token_0.balanceOf(yas_pool.contract_address);
+                let balance_token_1 = token_1.balanceOf(yas_pool.contract_address);
+
+                assert(balance_token_0 == 9996, 'wrong balance token 0');
+                assert(balance_token_1 == 1000, 'wrong balance token 1');
+            }
+
+
+            #[test]
+            #[available_gas(200000000)]
+            fn test_initial_tick() {
+                let (yas_pool, token_0, token_1, mint_callback, min_tick, max_tick) = setup();
+                let (expected_min_tick, expected_max_tick) = get_min_tick_and_max_tick();
```

Comment:
What is the purpose of having this tuple (expected_min_tick, expected_max_tick)?